drivers/ever-hid.c

```diff
@@ -0,0 +1,638 @@
+/* ever-hid.c - subdriver to monitor EVER USB/HID devices with NUT
+ *
+ *  Copyright (C)
+ *  2003 - 2012	Arnaud Quette <ArnaudQuette@Eaton.com>
+ *  2005 - 2006	Peter Selinger <selinger@users.sourceforge.net>
+ *  2008 - 2009	Arjen de Korte <adkorte-guest@alioth.debian.org>
+ *  2013 Charles Lepple <clepple+nut@gmail.com>
+ *
+ *  Note: this subdriver was initially generated as a "stub" by the
+ *  gen-usbhid-subdriver script. It must be customized.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "usbhid-ups.h"
+#include "ever-hid.h"
+#include "main.h"	/* for getval() */
+#include "usb-common.h"
+
+#define EVER_HID_VERSION	"Ever HID 0.1"
+/* FIXME: experimental flag to be put in upsdrv_info */
+
+/* Ever */
+#define EVER_VENDORID	0x2E51
+
+/* USB IDs device table */
+static usb_device_id_t ever_usb_device_table[] = {
+
+	{ USB_DEVICE(0x0483, 0xa113), NULL },
```

Comment:
Yes, this is correct. 0x0483  is assigned to STMicroelectronics.  We obtained PID number 0x0a113 from STMicroelectronics.